app/userdata.js

```diff
@@ -0,0 +1,33 @@
+const mysql = require('mysql');
+const axios = require('axios');
+const dbconfig = require('../dbconfig.json');
+const util = require('util')
+
+const db = wrapDB(dbconfig)
+
+
+
+function wrapDB(dbconfig) {
+    const pool = mysql.createPool(dbconfig)
+    return {
+      query(sql, args) {
+        console.log("in query in wrapper")
+        return util.promisify(pool.query)
+          .call(pool, sql, args)
+      },
+      release() {
+        return util.promisify(pool.releaseConnection)
+          .call(pool)
+      }
+    }
+  }
+  // Check this method 
+  exports.createUser = async (newUser) => {
+    try {
+      const usersResponse = await axios.post('http://localhost:8080/api/registeruser', { email: newUser.email, password: newUser.password, role: newUser.role })
```

Comment:
you're passing the password in plaintext over http here, you need to encrypt either your transmission (using https) or your password before sending

Reply:
We've got the functionality for HTTPS, but it is using a locally generated key that hasn't been verified, would it be better to swap entirely to an encryption before transmission approach?